packages/core/src/ctype/CType.schemas.ts

```diff
@@ -66,6 +49,89 @@ export const CTypeModelV1: JsonSchema.Schema & { $id: string } = {
     'title',
     'type',
   ],
+  definitions: {
+    cTypeReference: {
+      additionalProperties: false,
+      properties: {
+        $ref: {
+          pattern: '^kilt:ctype:0x[0-9a-f]+(#/properties/.+)?$',
+          format: 'uri',
+          type: 'string',
+        },
+      },
+      required: ['$ref'],
+    },
+    string: {
+      additionalProperties: false,
+      properties: {
+        type: {
+          const: 'string',
+        },
+        format: { enum: ['date', 'time', 'uri'] },
+        enum: {
+          type: 'array',
+          items: { type: 'string' },
+        },
+        minLength: {
+          type: 'number',
+        },
+        maxLength: {
+          type: 'number',
+        },
+      },
+      required: ['type'],
+    },
+    boolean: {
+      additionalProperties: false,
+      properties: {
+        type: {
+          const: 'boolean',
+        },
+      },
+      required: ['type'],
+    },
+    number: {
+      additionalProperties: false,
+      properties: {
+        type: {
+          enum: ['integer', 'number'],
+        },
+        enum: {
+          type: 'array',
+          items: { type: 'number' },
+        },
+        minimum: {
+          type: 'number',
+        },
+        maximum: {
+          type: 'number',
+        },
+      },
+      required: ['type'],
+    },
+    array: {
+      additionalProperties: false,
+      properties: {
+        type: { const: 'array' },
+        items: {
+          oneOf: [
+            { $ref: '#/definitions/string' },
+            { $ref: '#/definitions/number' },
+            { $ref: '#/definitions/boolean' },
+            // TODO: do we have reasons not to allow CType references in arrays?
```

Comment:
Do we have reasons not to allow arrays and objects inside arrays?

Reply:
Ok, so we just allow for 1 layer of nesting at the CType level, and anything else would be split up into nested CType structures, correct?

Reply:
Exactly